R/one_call_pipe_linter.R

```diff
@@ -0,0 +1,63 @@
+#' Block single-call magrittr pipes
+#'
+#' Prefer using a plain call instead of a pipe with only one call,
+#'   i.e. `1:10 %>% sum()` should instead be `sum(1:10)`. Note that
+#'   calls in the first `%>%` argument count. `rowSums(x) %>% max()` is OK
+#'   because there are two total calls (`rowSums()` and `max()`).
+#'
+#' Note also that un-"called" steps are *not* counted, since they should
+#'   be calls (see [pipe_call_linter()]).
+#'
+#' @evalRd rd_tags("one_call_pipe_linter")
+#' @seealso
+#' - [linters] for a complete list of linters available in lintr.
+#' - <https://style.tidyverse.org/pipes.html#short-pipes>
```

Comment:
Means we should make it a default linter, no?

Reply:
Not totally clear... internally our discussion put a lot of weight on **consider** here

> A one-step pipe can stay on one line, but unless you plan to expand it later on, you should **consider** rewriting it to a regular function call.

So there's some ambiguity in the style guide on whether to _always_ enforce this, or even if it should be removed _most_ of the time. We found the `# nolint` workaround too disruptive because there's so many false positives.

OTOH our rules are not identical to the Tidyverse rules, and we are free to continue disabling it. So {lintr}'s decision should ultimately be independent.

cc @IndrajeetPatil here.

Reply:
Okay let's keep it non-default then.